templates/tonic-serviceaccount-bindings.yaml

```diff
@@ -0,0 +1,43 @@
+{{- define "kubernetes.rbac.binding" }}
+{{- $top := first . }}
+{{- $kind := index . 1}}
+{{- $serviceAccountName := index . 2 }}
+{{- $role := index . 3 }}
+{{- $labels := index . 4 }}
+{{- $annotations := index . 5 }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: {{ $kind }}
+metadata:
+  annotations:
+    {{- include "tonic.annotations" (list $top $annotations) | nindent 4 }}
+  labels:
+    {{- include "tonic.allLabels" (list $top $labels) | nindent 4 }}
+  name: {{ $serviceAccountName }}-{{ $role.name }}
+  namespace: {{ $top.Release.Namespace }}
+roleRef:
+  {{- (toYaml $role) | nindent 2 }}
+subjects:
+- kind: ServiceAccount
+  name: {{ $serviceAccountName }}
+  namespace: {{ $top.Release.Namespace }}
+{{- end }}
+
+{{- define "tonic.serviceAccount.bindings" }}
+{{- $top := first . }}
+{{- $kind := index . 1 }}
+{{- $serviceAccountName := index . 2 }}
+{{- $definition := index . 3 }}
+{{- if and $definition $definition.bindings }}
+{{- $labels := $definition.labels }}
+{{- $annotations := $definition.annotations }}
+{{- range $role := $definition.bindings }}
+{{ include "kubernetes.rbac.binding" (list $top $kind $serviceAccountName $role $labels $annotations) }}
+---
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- $serviceAccountName := (include "tonic.serviceAccountName" .) }}
+{{- include "tonic.serviceAccount.bindings" (list $ "RoleBinding" $serviceAccountName .Values.serviceAccount.roleBindings) }}
+{{- include "tonic.serviceAccount.bindings" (list $ "ClusterRoleBinding" $serviceAccountName .Values.serviceAccount.clusterRoleBindings) }}
+
```

Comment:
this is a level of helm I have not been exposed to yet. Did you write this or is it taken from a known source?

Very cool either way

Reply:
artisan yaml